### ISSUE TITLE
Change handling of dashboard `pin` params for interactive use

### ISSUE THREAD

**Issue by juliasilge** [rstudio/vetiver-r#107]
In #98 we added the R Markdown template for a model monitoring dashboard. Thanks to @cderv we managed a way to get the `pins` parameters into the knitting environment, via the tiny little function:

```r
pins <- get_vetiver_dashboard_pins()
```

As @cderv outlined, one downside of this is that the parameters like `pins$board` and friends are not available in the environment during interactive use, like evaluating chunk by chunk; they are only available when evaluated.

I think this is OK for now, but later we may want to improve this approach.

**Comment by cderv** [comment]
Please, do ping me again when you want to deal with this. I can discuss with other R Markdown expert and IDE team to understand what would really be possible. 

Takes from our past discussion and my research: We could use `params` as you first intended, but we need a small adjustment to flexdashboard so that you can use the `params` value to create the `navbar` element you need. Currently you can't because the navbar element will be created when the format function is evaluated, which happens before knitting. 

Other solution to define the pins information in a setup chunk inside the RMD require the same adjustment. You can have something like 
````markdown
We are inside the Rmd

```{r}
pins <- use_vetiver_dashboard_pins(board, name, version)
```
````

This would be available interactively, and during knitting, but the navbar creation in flexdashboard would need to be after knitting so that you can use the pin information. 

If you are interested by one of this solution, I can PR into **flexdashboard** the change I have in mind.

**Comment by juliasilge** [comment]
Actually, that change to flexdashboard would be perfect @cderv; that would really work well.

Can you put that on your "to do" list, for when it works for your priorities? I don't think this necessarily needs to be before conf (what we have will work for now), but we would like to make this change eventually.

**Comment by cderv** [comment]
Yes I'll put that on my list and test the idea soon to confirm this is working effectively as I mentioned.

**Comment by cderv** [comment]
@juliasilge How would you see your vetiver dashboard configuration if we were to use params inside in the document ? 
Is this the same than in #98 ? or as this change because package has evolved ? 

I have done the tweak to define navbar later in the process at https://github.com/cderv/flexdashboard/tree/navbar-in-preprocess

This means that the navbar will be created after knitr evalutation in the document, but I don't remember exactly how you wanted to tweak it. Possibly using the solution 2 mentioned here https://github.com/rstudio/vetiver-r/issues/107#issuecomment-1154873577 ? 

Also, we are adding some new feature in **rmarkdown** in https://github.com/rstudio/rmarkdown/pull/2462 to be able to modify some part of the output format from within the document. maybe it can be useful 🤔  but not sure yet.

**Comment by juliasilge** [comment]
@cderv This isn't new or different compared to what we discussed in #98, but specifically I was still interested in solving the problem [you noticed here](https://github.com/rstudio/vetiver-r/pull/98#discussion_r895670486). I'd like users to be able to put their `board` / `name` / `version` in the YAML front matter (as params?) and have those values available _both_ to render the navbar _and_ for interactive use in the IDE, like params are typically available when working with parameterized R Markdown.

You can see the results of the current navbar behavior here:
https://colorado.posit.co/rsc/seattle-housing-dashboard/

To get this to work starting from the template, change the YAML front matter to:

```yaml
  vetiver::vetiver_dashboard:
    pins:
      board: !expr pins::board_connect()  ## need credentials saved in .Renviron for colorado
      name: 'julia.silge/seattle_rf'
      version: NULL
```
